mypy/types.py

```diff
@@ -783,6 +784,30 @@ def deserialize(self, data: JsonDict) -> 'EllipsisType':
         return EllipsisType()
 
 
+class TypeType(Type):
+    """For types like Type[User].
+
+    This annotates variables that are class objects, constrained by
+    the type argument.  See PEP 484 for more details.
```

Comment:
Thanks for the review! I pushed a new version that addresses some of these
(though for a few of the tougher ones I added an XXX comment -- I need more
time to iterate on these).